web3swift/src/SIWE/SiweVerifier.swift

```diff
@@ -0,0 +1,92 @@
+//
+//  SiweVerifier.swift
+//  
+//
+//  Created by Rodrigo Kreutz on 14/06/22.
+//
+
+import Foundation
+import BigInt
+
+/// An object which will verify if a given `SiweMessage` and signature match with the EVM address provided
+public class SiweVerifier {
+
+    /// Errors thrown when verifing a given message agains a signature
+    public enum Error: Swift.Error {
+        /// The provided message is from a different network than the client's.
+        case differentNetwork
+        /// The provided message has a `notBefore` field set and the verification time is before that time
+        case messageIsNotActiveYet
+        /// The provided message has a `expirationTime` field set and the verification time is after or at that time
+        case messageIsExpired
+        /// Failed to fetch `utf8` data from the message string
+        case invalidMessageData
+        /// Failed to fetch the hex data from the signature
+        case invalidSignature
+    }
+
+    private let client: EthereumClientProtocol
+    private let dateProvider: () -> Date
+
+    public init(client: EthereumClientProtocol, dateProvider: @escaping () -> Date = Date.init) {
+        self.client = client
+        self.dateProvider = dateProvider
+    }
+
+    /// Verifies if a given EIP-4361 string message was signed by the address in the message.
+    /// - Parameters:
+    ///   - message: the EIP-4361 string message
+    ///   - signature: the hexadecimal string of the alleged signature, prefixed with `0x`
+    /// - Returns: a `Bool` indicating if the pair message-signature is verified (whether or not the signature came from the address in the message)
+    /// - Throws: any errors thrown from `SiweMessage.init(_:)` and `SiweVerifier.verify(message:against)`
+    public final func verify(_ message: String, against signature: String) async throws -> Bool {
+        return try await verify(message: SiweMessage(message), against: signature)
+    }
+
+    /// Verifies if a given `SiweMessage` was signed by the address in the message.
+    /// - Parameters:
+    ///   - message: the message to be verified
+    ///   - signature: the hexadecimal string of the alleged signature, prefixed with `0x`
+    /// - Returns: a `Bool` indicating if the pair message-signature is verified (whether or not the signature came from the address in the message)
+    /// - Throws: `SiweVerifier.Error` if message is not verifiable;
+    ///           might throw `KeyUtilError` in case recovering the address that signed the message fails
+    open func verify(message: SiweMessage, against signature: String) async throws -> Bool {
```

Comment:
I think this should be `public` not `open`?

Reply:
Done [39e0169](https://github.com/argentlabs/web3.swift/pull/246/commits/39e0169c91379493584212e8314f29c75b3f7caa)